From 4e8607f3affa99483829395c9647169b06653e75 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Fri, 31 Jan 2014 11:34:24 +0100
Subject: [PATCH 095/182] watchdog: xilinx: Fix logic around enable-once
 feature

https://github.com/analogdevicesinc/linux.git xcomm_zynq
commit 7d591a7689c42bc39c046ee4e41b5c6c73845745

If xlnx,wdt-enable-once is specified and it is non zero
then setup WDOG_NO_WAY_OUT flag to prevert calling
stop function.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
(cherry picked from commit a8ada939200bd450ee0335421fc4aba1177e14f3)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/watchdog/of_xilinx_wdt.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/drivers/watchdog/of_xilinx_wdt.c b/drivers/watchdog/of_xilinx_wdt.c
index 2b0bdab..7948def 100644
--- a/drivers/watchdog/of_xilinx_wdt.c
+++ b/drivers/watchdog/of_xilinx_wdt.c
@@ -182,10 +182,9 @@ static int xwdt_probe(struct platform_device *pdev)
 		no_timeout = true;
 	}
 
-	/* FIXME this is weird */
 	rc = of_property_read_u32(pdev->dev.of_node, "xlnx,wdt-enable-once",
 				  &enable_once);
-	if (rc || enable_once) {
+	if (!rc && enable_once) {
 		dev_warn(&pdev->dev,
 			 "Parameter \"xlnx,wdt-enable-once\" not found\n");
 		watchdog_set_nowayout(xilinx_wdt_wdd, true);
-- 
1.7.5.4

